avalon/pipeline.py

```diff
@@ -1313,6 +1335,67 @@ def get_repres_contexts(representation_ids, dbcon=None):
     return contexts
 
 
+def get_subset_contexts(subset_ids, dbcon=None):
+    """Return parenthood context for subset.
+
+    Args:
+        subset_ids (list): The subset ids.
+        dbcon (AvalonMongoDB): Mongo connection object. `avalon.io` used when
+            not entered.
+
+    Returns:
+        dict: The full representation context by representation id.
+
+    """
+    if not dbcon:
+        dbcon = io
+
+    contexts = {}
+    if not subset_ids:
+        return contexts
+
+    _subset_ids = []
+    for id in subset_ids:
+        if isinstance(id, six.string_types):
+            id = io.ObjectId(id)
+        _subset_ids.append(id)
+
+    subset_docs = dbcon.find({
+        "type": "subset",
+        "_id": {"$in": list(subset_ids)}
+    })
+    subset_docs_by_id = {}
+    asset_ids = set()
+    for subset_doc in subset_docs:
+        subset_docs_by_id[subset_doc["_id"]] = subset_doc
+        asset_ids.add(subset_doc["parent"])
+
+    asset_docs = dbcon.find({
+        "type": "asset",
+        "_id": {"$in": list(asset_ids)}
+    })
+    asset_docs_by_id = {
+        asset_doc["_id"]: asset_doc
+        for asset_doc in asset_docs
+    }
+
+    project_doc = dbcon.find_one({"type": "project"})
+
+    for id, doc in subset_docs_by_id.items():
+        asset_doc = asset_docs_by_id[subset_doc["parent"]]
```

Comment:
Should be `asset_docs_by_id[doc["parent"]]` or maybe rather change loop variable names `id` -> `subset_id` (as `id` in python builtin function) and `doc` to `subset_doc` to say what it is :)

Reply:
Good catch! Fixed.